ledger/participant-integration-api/src/main/scala/platform/index/InMemoryStateUpdater.scala

```diff
@@ -100,14 +93,42 @@ private[platform] object InMemoryStateUpdater {
     updateCachesExecutionContext = ExecutionContext.fromExecutorService(updateCachesExecutor),
     metrics = metrics,
   )(
-    updateCaches = updateCaches(inMemoryState),
-    convertTransactionAccepted = convertTransactionAccepted,
-    convertTransactionRejected = convertTransactionRejected,
-    updateLedgerEnd = updateLedgerEnd(inMemoryState),
+    prepare = prepare,
+    update = update(inMemoryState, loggingContext),
   )
 
-  private def updateCaches(inMemoryState: InMemoryState)(
-      updates: Vector[TransactionLogUpdate]
+  def combinePackageMetadata(batch: Vector[(Offset, Update)]): PackageMetadata =
+    batch.foldLeft(PackageMetadata()) {
+      case (metadata, (_, uploadEvent: Update.PublicPackageUpload)) =>
+        uploadEvent.archives.foldLeft(metadata) { case (acc, archive) =>
+          acc.append(PackageMetadata.from(archive))
+        }
+      case (metadata, _) => metadata
+    }
+
+  def prepare(batch: Vector[(Offset, Update)], lastEventSequentialId: Long): PrepareResult =
+    PrepareResult(
+      logUpdate = batch.collect {
+        case (offset, u: Update.TransactionAccepted) => convertTransactionAccepted(offset, u)
+        case (offset, u: Update.CommandRejected) => convertTransactionRejected(offset, u)
+      },
+      lastOffset = batch.last._1,
+      lastEventSequentialId = lastEventSequentialId,
+      packageMetadata = combinePackageMetadata(batch),
+    )
+
+  def update(
+      inMemoryState: InMemoryState,
+      loggingContext: LoggingContext,
+  )(result: PrepareResult): Unit = {
+    inMemoryState.packageMetadataView.update(result.packageMetadata)
+    updateCaches(inMemoryState, result.logUpdate)
+    updateLedgerEnd(inMemoryState, result.lastOffset, result.lastEventSequentialId)(loggingContext)
```

Comment:
please add a comment here that this needs to come last in this method and the reason for it

Reply:
there is such a comment already inside `updateLedgerEnd` i will add another one on top of it